src/contexts/Analytics/Analytics.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+  useContext,
+  useCallback,
+  useRef,
+  createContext,
+  FunctionComponent,
+} from "react";
+import { useLocation } from "react-router-dom";
+import { PageViewConfig } from "./types";
+
+export interface AnalyticsAPI {
+  trackPageView: (opts: PageViewConfig) => void;
+}
+
+const AnalyticsContext = createContext<AnalyticsAPI>({
+  trackPageView: () => {},
+});
+
+export const useAnalytics = () => useContext(AnalyticsContext);
+
+export const AnalyticsProvider: FunctionComponent = ({ children }) => {
+  const { pathname } = useLocation();
+
+  const prevPath = useRef<null | string>(null);
```

Comment:
In order to prevent duplicate tracking calls, we want to store the previous location to guarantee we don't somehow fire a pageview call twice for the same url